src/core/Table/Table.tsx

```diff
@@ -149,6 +183,19 @@ export type TableProps<
    * @default true
    */
   selectRowOnClick?: boolean;
+  /**
+   * Function that returns pagination component. Recommended to use `TablePaginator`.
+   * @example
+   * (props: TablePaginatorRendererProps) => (
+   *   <TablePaginator {...props} />
+   * )
+   */
+  paginatorRenderer?: (props: TablePaginatorRendererProps) => React.ReactNode;
```

Comment:
so if you're sticking with this approach, I would recommend adding more details in the comment of this prop and the comments of the individual props.